services/shotgrid_common/utils.py

```diff
@@ -1660,9 +1661,18 @@ def _get_sg_note(sg_note_id, sg_session):
             "sg_ayon_id",
             "user",
             "note_links",
-            "addressings_to"
+            "addressings_to",
+            "attachments"
         ]
     )
+    if sg_note.get("attachments"):
+        # download attachments locally
+        for attachment in sg_note["attachments"]:
+            with tempfile.NamedTemporaryFile(
+                delete=False,
+                suffix=attachment["name"]
+            ) as tmp_file:
+                attachment["local_path"] = sg_session.download_attachment(attachment, file_path=tmp_file.name)
```

Comment:
I'd suggest to wrap it inside a try-except, we never know what can happen when downloading stuff (server timeout, permission error...).

```suggestion
    if sg_note.get("attachements"):
        # download attachments locally
        for attachment in sg_note["attachments"]:
            try:
                with tempfile.NamedTemporaryFile(
                    delete=False,
                    suffix=attachment["name"]
                ) as tmp_file:
                    attachment["local_path"] = sg_session.download_attachment(attachment, file_path=tmp_file.name)

            except Exception as error:
                log.warning(
                    "Could not download note attachement for %r (attachement: %r): %r.",
                    sg_note,
                    attachment,
                    error
                )
```